File: SDKs/amazonaws.com/rds/2014-09-01/go/pkg/models/operations/getrebootdbinstance.go
package operations




type GetRebootDbInstanceActionEnum string

const (
    GetRebootDbInstanceActionEnumRebootDbInstance GetRebootDbInstanceActionEnum = "RebootDBInstance"
)



type GetRebootDbInstanceVersionEnum string

const (
    GetRebootDbInstanceVersionEnumTwoThousandAndFourteen0901 GetRebootDbInstanceVersionEnum = "2014-09-01"
)


type GetRebootDbInstanceQueryParams struct {
    Action GetRebootDbInstanceActionEnum `queryParam:"style=form,explode=true,name=Action"`
    DbInstanceIdentifier string `queryParam:"style=form,explode=true,name=DBInstanceIdentifier"`
    ForceFailover *bool `queryParam:"style=form,explode=true,name=ForceFailover"`
    Version GetRebootDbInstanceVersionEnum `queryParam:"style=form,explode=true,name=Version"`
    
}

type GetRebootDbInstanceHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    
}

type GetRebootDbInstanceRequest struct {
    QueryParams GetRebootDbInstanceQueryParams 
    Headers GetRebootDbInstanceHeaders 
    
}

type GetRebootDbInstanceResponse struct {
    Body []byte 
    ContentType string 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/rds/2013-01-10/typescript/src/sdk/models/operations/getrebootdbinstance.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum GetRebootDbInstanceActionEnum {
    RebootDbInstance = "RebootDBInstance"
}

export enum GetRebootDbInstanceVersionEnum {
    TwoThousandAndThirteen0110 = "2013-01-10"
}


export class GetRebootDbInstanceQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: GetRebootDbInstanceActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=DBInstanceIdentifier" })
  dbInstanceIdentifier: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=ForceFailover" })
  forceFailover?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: GetRebootDbInstanceVersionEnum;
}


export class GetRebootDbInstanceHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class GetRebootDbInstanceRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetRebootDbInstanceQueryParams;

  @SpeakeasyMetadata()
  headers: GetRebootDbInstanceHeaders;
}


export class GetRebootDbInstanceResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
